Added support (and style) for other files ext.

edgardpachper opened the pull request:
added support for audio files (.opus, .mp3, .aac, .wav) 
added support for video files (.mp4, .mpeg, .3gp, .avi, .wmm) 
added support for image files (jpg, .jpeg, .png, .tiff, .ico) 
added style for audio, video, images files.

after deleting the <li> tags.

edgardpachper committed:
Added support (and style) for other files ext.

added support for audio files (.opus, .mp3, .aac, .wav)
added support for video files (.mp4, .mpeg, .3gp, .avi, .wmm)
added support for image files (jpg, .jpeg, .png, .tiff, .ico)
added style for audio, video, images files.

automaciej commented on whatsapp_archive.py:
```diff
@@ -271,32 +271,18 @@ def FormatHTML(data):
                 {% if media is not none %}
-                    {% if "IMG" in media %}
-                        <li>
+                    {% if "IMG" in media or "jpeg" in media or "png" in media or "tiff" in media or "ico" in media or "gif" in media %}
```
This doesn't seem to scale well. It should probably be a list in the template data, and here there should be just one expression. Something like `{% if media.IsImage() %}` or `{% if media.type in image_types %}` or something readable like that.

Also, what if there's a file called "jpeg.gif", will your code work correctly?

automaciej commented on whatsapp_archive.py:
```diff
@@ -271,32 +271,18 @@ def FormatHTML(data):
-                          <audio controls>
+                    {% elif "opus" in media or "m4a" in media or "mp3" in media or "AAC" in media or "WAV" in media %}
+                        <audio controls>
                             <source src="{{ media }}" type="audio/x-m4a">
```
Shouldn't the type here should match the type detected in line 276?

automaciej commented on whatsapp_archive.py:
```diff
@@ -271,32 +271,18 @@ def FormatHTML(data):
+                        </audio>
+                    {% elif "mp4" in media or ".MPEG" in media or ".3GP" in media or ".AVI" in media or ".WMM" in media %}
+                       <video controls>
+                         <source src="{{ media }}" type="video/mp4"/>
```
Similar issue here, the `type` should match the type detected in line 280.

automaciej commented on whatsapp_archive.py:
```diff
@@ -271,32 +271,18 @@ def FormatHTML(data):
+                         <source src="{{ media }}" type="video/mp4"/>
+                       </video>
                     {% else %}
-                        <li>
```
Is this still valid HTML? I'm pretty sure that there can't be content directly under `<ul>`, I mean other than `<li>` elements.

automaciej committed:
Merge branch 'master' into formats

automaciej committed:
Remove an unnecessary print.

automaciej committed:
Add a Media dataclass and use mimetypes.

I'm using dataclasses just so I can use a short syntax in the modules.
Thankfully Python 3 can guess mime types for us, so I don't need to
write any of that code.
